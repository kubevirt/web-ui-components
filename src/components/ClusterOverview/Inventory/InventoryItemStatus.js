import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'patternfly-react';

const StatusPropTypes = {
  count: PropTypes.number.isRequired,
};

const OkStatus = ({ count }) => (
  <div className="kubevirt-inventory__row-status-item">
    <Icon type="fa" name="check-circle" size="2x" className="kubevirt-inventory__row-status-item-icon--ok" />
    <span className="kubevirt-inventory__row-status-item-text">{count}</span>
  </div>
);

OkStatus.propTypes = StatusPropTypes;

const WarnStatus = ({ count }) => (
  <div className="kubevirt-inventory__row-status-item">
    <Icon type="fa" name="exclamation-triangle" size="2x" className="kubevirt-inventory__row-status-item-icon--warn" />
    <span className="kubevirt-inventory__row-status-item-text">{count}</span>
  </div>
);

WarnStatus.propTypes = StatusPropTypes;

const ErrorStatus = ({ count }) => (
  <div className="kubevirt-inventory__row-status-item">
    <Icon type="fa" name="exclamation-circle" size="2x" className="kubevirt-inventory__row-status-item-icon--error" />
    <span className="kubevirt-inventory__row-status-item-text">{count}</span>
  </div>
);

ErrorStatus.propTypes = StatusPropTypes;

const InProgressStatus = ({ count }) => (
  <div className="kubevirt-inventory__row-status-item">
    <Icon type="pf" name="in-progress" size="2x" />
    <span className="kubevirt-inventory__row-status-item-text">{count}</span>
  </div>
);

InProgressStatus.propTypes = StatusPropTypes;

const Status = ({ Component, count, ...props }) => count > 0 && <Component count={count} {...props} />;

export const InventoryItemStatus = ({ ok, warn, error, inProgress }) => (
  <div className="kubevirt-inventory__row-status">
    <Status Component={InProgressStatus} count={inProgress} />
    <Status Component={ErrorStatus} count={error} />
    <Status Component={WarnStatus} count={warn} />
    <Status Component={OkStatus} count={ok} />
  </div>
);

InventoryItemStatus.defaultProps = {
  ok: 0,
  warn: 0,
  error: 0,
  inProgress: 0,
};

InventoryItemStatus.propTypes = {
  ok: PropTypes.number,
  warn: PropTypes.number,
  error: PropTypes.number,
  inProgress: PropTypes.number,
};
