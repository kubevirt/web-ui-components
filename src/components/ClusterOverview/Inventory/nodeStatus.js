import { get } from 'lodash';

export const getNodeErrorStatuses = node => {
  const conditions = get(node, 'status.conditions');
  const errors = conditions
    .filter(c => c.type !== 'Ready')
    .filter(c => c.status === 'True')
    .map(c => ({
      type: c.type,
      message: c.message,
    }));
  if (errors.length > 0) {
    const ready = conditions.find(c => c.type === 'Ready');
    if (ready.status === 'False') {
      errors.push({
        type: ready.type,
        message: ready.message,
      });
    }
  }
  return errors;
};
