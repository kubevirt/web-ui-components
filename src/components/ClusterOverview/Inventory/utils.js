import { getVmStatus, VM_STATUS_ALL_ERROR, VM_STATUS_ALL_PROGRESS } from '../../../utils/status/vm';
import { getPodStatus, POD_STATUS_ALL_ERROR, POD_STATUS_ALL_PROGRESS } from '../../../utils/status/pod';
import { getNodeStatus, NODE_STATUS_ALL_ERROR, NODE_STATUS_ALL_WARN } from '../../../utils/status/node';
import { getPvcStatus, PVC_STATUS_ALL_ERROR, PVC_STATUS_ALL_PROGRESS } from '../../../utils/status/pvc';

// same as InventoryItemStatus props
const STATUS_RESULT_OK = 'ok';
const STATUS_RESULT_WARN = 'warn';
const STATUS_RESULT_ERROR = 'error';
const STATUS_RESULT_IN_PROGRESS = 'inProgress';

const mapStatuses = (entities, mapEntityToStatusResult) => {
  const result = {
    [STATUS_RESULT_OK]: 0,
    [STATUS_RESULT_WARN]: 0,
    [STATUS_RESULT_ERROR]: 0,
    [STATUS_RESULT_IN_PROGRESS]: 0,
    count: null,
  };

  if (entities) {
    entities.forEach(entity => {
      const resolvedStatusResult = (mapEntityToStatusResult && mapEntityToStatusResult(entity)) || STATUS_RESULT_OK;
      result[resolvedStatusResult]++;
    });
    result.count = entities.length;
  }

  return result;
};

const resolveStatusResult = (status, mapper) =>
  Object.keys(mapper).find(key => mapper[key].includes(status)) || STATUS_RESULT_OK;

export const mapNodesToProps = nodes =>
  mapStatuses(nodes, node =>
    resolveStatusResult(getNodeStatus(node).status, {
      [STATUS_RESULT_ERROR]: NODE_STATUS_ALL_ERROR,
      [STATUS_RESULT_WARN]: NODE_STATUS_ALL_WARN,
    })
  );

export const mapVmsToProps = (vms, pods, migrations) =>
  mapStatuses(vms, vm =>
    resolveStatusResult(getVmStatus(vm, pods, migrations).status, {
      [STATUS_RESULT_ERROR]: VM_STATUS_ALL_ERROR,
      [STATUS_RESULT_IN_PROGRESS]: VM_STATUS_ALL_PROGRESS,
    })
  );

export const mapPvcsToProps = pvcs =>
  mapStatuses(pvcs, pvc =>
    resolveStatusResult(getPvcStatus(pvc).status, {
      [STATUS_RESULT_ERROR]: PVC_STATUS_ALL_ERROR,
      [STATUS_RESULT_IN_PROGRESS]: PVC_STATUS_ALL_PROGRESS,
    })
  );

export const mapPodsToProps = pods =>
  mapStatuses(pods, pod =>
    resolveStatusResult(getPodStatus(pod).status, {
      [STATUS_RESULT_ERROR]: POD_STATUS_ALL_ERROR,
      [STATUS_RESULT_IN_PROGRESS]: POD_STATUS_ALL_PROGRESS,
    })
  );
