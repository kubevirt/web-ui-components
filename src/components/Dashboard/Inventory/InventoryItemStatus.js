import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'patternfly-react';

const StatusPropTypes = {
  count: PropTypes.number.isRequired,
};

const OkStatus = ({ count }) => (
  <div className="kubevirt-inventory__row-status-item">
    <Icon
      type="fa"
      name="check-circle"
      className="kubevirt-inventory__row-status-item-icon--ok kubevirt-dashboard__icon-sm"
    />
    <span className="kubevirt-inventory__row-status-item-text">{count}</span>
  </div>
);

OkStatus.propTypes = StatusPropTypes;

const WarnStatus = ({ count }) => (
  <div className="kubevirt-inventory__row-status-item">
    <Icon
      type="fa"
      name="exclamation-triangle"
      className="kubevirt-inventory__row-status-item-icon--warn kubevirt-dashboard__icon-sm"
    />
    <span className="kubevirt-inventory__row-status-item-text">{count}</span>
  </div>
);

WarnStatus.propTypes = StatusPropTypes;

const ErrorStatus = ({ count }) => (
  <div className="kubevirt-inventory__row-status-item">
    <Icon
      type="fa"
      name="exclamation-circle"
      className="kubevirt-inventory__row-status-item-icon--error kubevirt-dashboard__icon-sm"
    />
    <span className="kubevirt-inventory__row-status-item-text">{count}</span>
  </div>
);

ErrorStatus.propTypes = StatusPropTypes;

const InProgressStatus = ({ count }) => (
  <div className="kubevirt-inventory__row-status-item">
    <Icon
      type="pf"
      name="in-progress"
      className="kubevirt-inventory__row-status-item-icon--in-progress kubevirt-dashboard__icon-sm"
    />
    <span className="kubevirt-inventory__row-status-item-text">{count}</span>
  </div>
);

InProgressStatus.propTypes = StatusPropTypes;

const OffStatus = ({ count }) => (
  <div className="kubevirt-inventory__row-status-item">
    <Icon type="pf" name="off" className="kubevirt-inventory__row-status-item-icon--off" />
    <span className="kubevirt-inventory__row-status-item-text">{count}</span>
  </div>
);

OffStatus.propTypes = StatusPropTypes;

const Status = ({ Component, count, ...props }) => count > 0 && <Component count={count} {...props} />;

export const InventoryItemStatus = ({ ok, warn, error, inProgress, off }) => (
  <div className="kubevirt-inventory__row-status">
    <Status Component={InProgressStatus} count={inProgress} />
    <Status Component={ErrorStatus} count={error} />
    <Status Component={WarnStatus} count={warn} />
    <Status Component={OffStatus} count={off} />
    <Status Component={OkStatus} count={ok} />
  </div>
);

InventoryItemStatus.defaultProps = {
  ok: 0,
  warn: 0,
  error: 0,
  inProgress: 0,
  off: 0,
};

InventoryItemStatus.propTypes = {
  ok: PropTypes.number,
  warn: PropTypes.number,
  error: PropTypes.number,
  inProgress: PropTypes.number,
  off: PropTypes.number,
};
