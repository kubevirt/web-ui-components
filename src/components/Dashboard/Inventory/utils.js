import {
  getVmStatus,
  VM_STATUS_ALL_WARNING,
  VM_STATUS_ALL_ERROR,
  VM_STATUS_ALL_PROGRESS,
  VM_STATUS_ALL_OFF,
} from '../../../utils/status/vm';
import { getPodStatus, POD_STATUS_ALL_ERROR, POD_STATUS_ALL_PROGRESS } from '../../../utils/status/pod';
import { getNodeStatus, NODE_STATUS_ALL_ERROR, NODE_STATUS_ALL_WARN } from '../../../utils/status/node';
import { getPvcStatus, PVC_STATUS_ALL_ERROR, PVC_STATUS_ALL_PROGRESS } from '../../../utils/status/pvc';
import {
  getHostStatus,
  HOST_STATUS_ALL_ERROR,
  HOST_STATUS_ALL_WARN,
  HOST_STATUS_ALL_PROGRESS,
} from '../../../utils/status/host';
import { getPvStatus, PV_STATUS_ALL_ERROR, PV_STATUS_ALL_PROGRESS } from '../../../utils/status/pv';

import { getCapacityStats } from '../../../selectors';

// same as InventoryItemStatus props
export const STATUS_RESULT_OK = 'ok';
export const STATUS_RESULT_WARN = 'warn';
export const STATUS_RESULT_ERROR = 'error';
export const STATUS_RESULT_IN_PROGRESS = 'inProgress';
export const STATUS_RESULT_OFF = 'off';

const mapStatuses = (entities, mapEntityToStatusResult) => {
  const result = {
    [STATUS_RESULT_OK]: 0,
    [STATUS_RESULT_WARN]: 0,
    [STATUS_RESULT_ERROR]: 0,
    [STATUS_RESULT_IN_PROGRESS]: 0,
    [STATUS_RESULT_OFF]: 0,
    count: null,
  };

  if (entities) {
    entities.forEach(entity => {
      const resolvedStatusResult = (mapEntityToStatusResult && mapEntityToStatusResult(entity)) || STATUS_RESULT_OK;
      result[resolvedStatusResult]++;
    });
    result.count = entities.length;
  }

  return result;
};

const resolveStatusResult = (status, mapper) =>
  Object.keys(mapper).find(key => mapper[key].includes(status)) || STATUS_RESULT_OK;

export const mapNodesToProps = nodes =>
  mapStatuses(nodes, node =>
    resolveStatusResult(getNodeStatus(node).status, {
      [STATUS_RESULT_ERROR]: NODE_STATUS_ALL_ERROR,
      [STATUS_RESULT_WARN]: NODE_STATUS_ALL_WARN,
    })
  );

export const mapVmsToProps = (vms, pods, migrations) =>
  mapStatuses(vms, vm =>
    resolveStatusResult(getVmStatus(vm, pods, migrations).status, {
      [STATUS_RESULT_WARN]: VM_STATUS_ALL_WARNING,
      [STATUS_RESULT_ERROR]: VM_STATUS_ALL_ERROR,
      [STATUS_RESULT_IN_PROGRESS]: VM_STATUS_ALL_PROGRESS,
      [STATUS_RESULT_OFF]: VM_STATUS_ALL_OFF,
    })
  );

export const mapPvcsToProps = pvcs =>
  mapStatuses(pvcs, pvc =>
    resolveStatusResult(getPvcStatus(pvc).status, {
      [STATUS_RESULT_ERROR]: PVC_STATUS_ALL_ERROR,
      [STATUS_RESULT_IN_PROGRESS]: PVC_STATUS_ALL_PROGRESS,
    })
  );

export const mapPvsToProps = pvs =>
  mapStatuses(pvs, pv =>
    resolveStatusResult(getPvStatus(pv).status, {
      [STATUS_RESULT_ERROR]: PV_STATUS_ALL_ERROR,
      [STATUS_RESULT_IN_PROGRESS]: PV_STATUS_ALL_PROGRESS,
    })
  );

export const mapPodsToProps = pods =>
  mapStatuses(pods, pod =>
    resolveStatusResult(getPodStatus(pod).status, {
      [STATUS_RESULT_ERROR]: POD_STATUS_ALL_ERROR,
      [STATUS_RESULT_IN_PROGRESS]: POD_STATUS_ALL_PROGRESS,
    })
  );

export const mapHostsToProps = hosts =>
  mapStatuses(hosts, host =>
    resolveStatusResult(getHostStatus(host).status, {
      [STATUS_RESULT_ERROR]: HOST_STATUS_ALL_ERROR,
      [STATUS_RESULT_WARN]: HOST_STATUS_ALL_WARN,
      [STATUS_RESULT_IN_PROGRESS]: HOST_STATUS_ALL_PROGRESS,
    })
  );

export const mapDiskStatsToProps = (cephOsdUp, cephOsdDown) => {
  const result = {
    [STATUS_RESULT_OK]: 0,
    [STATUS_RESULT_ERROR]: 0,
    count: cephOsdUp && cephOsdDown ? 0 : null,
  };

  const cephOsdUpCount = getCapacityStats(cephOsdUp);
  const cephOsdDownCount = getCapacityStats(cephOsdDown);

  if (cephOsdUpCount || cephOsdDownCount) {
    result[STATUS_RESULT_OK] = cephOsdUpCount;
    result[STATUS_RESULT_ERROR] = cephOsdDownCount;
    result.count = cephOsdUpCount + cephOsdDownCount;
  }

  return result;
};
