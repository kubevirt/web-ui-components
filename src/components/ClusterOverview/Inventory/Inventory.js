import React from 'react';
import PropTypes from 'prop-types';
import { Col, Icon } from 'patternfly-react';

import { getNodeErrorStatuses } from './nodeStatus';
import {
  DashboardCard,
  DashboardCardBody,
  DashboardCardHeader,
  DashboardCardTitle,
  DashboardCardTitleHelp,
} from '../../Dashboard/DashboardCard';
import { ClusterOverviewContextGenericConsumer } from '../ClusterOverviewContext';

const InventoryItemStatus = ({ item }) => {
  let ok = true;
  if (item.kind === 'Node') {
    ok = !item.data.some(i => getNodeErrorStatuses(i).length > 0);
  }
  return ok ? (
    <div className="kubevirt-inventory__item-status">
      <Icon type="fa" name="check-circle" size="2x" className="kubevirt-inventory__item-status-icon" />
      <span>{item.data.length}</span>
    </div>
  ) : (
    <div>
      <span className="glyphicon glyphicon-remove-circle" />
    </div>
  );
};

InventoryItemStatus.propTypes = {
  item: PropTypes.object.isRequired,
};

const InventoryBody = ({ inventory }) =>
  Object.keys(inventory).map(key => {
    const item = inventory[key];
    return (
      <div key={key} className="kubevirt-inventory__item">
        <Col lg={9} md={9} sm={9} xs={9}>
          {item.data.length} {item.title}
        </Col>
        <Col lg={3} md={3} sm={3} xs={3}>
          <InventoryItemStatus item={item} />
        </Col>
      </div>
    );
  });

InventoryBody.propTypes = {
  inventory: PropTypes.object.isRequired,
};

export const Inventory = ({ inventory, loaded }) => (
  <DashboardCard>
    <DashboardCardHeader>
      <DashboardCardTitle>Cluster inventory</DashboardCardTitle>
      <DashboardCardTitleHelp>help for inventory</DashboardCardTitleHelp>
    </DashboardCardHeader>
    <DashboardCardBody className="kubevirt-inventory__body" isLoading={!loaded}>
      <InventoryBody inventory={inventory} />
    </DashboardCardBody>
  </DashboardCard>
);

Inventory.defaultProps = {
  loaded: false,
};

Inventory.propTypes = {
  inventory: PropTypes.object.isRequired,
  loaded: PropTypes.bool,
};

const InventoryConnected = () => (
  <ClusterOverviewContextGenericConsumer Component={Inventory} dataPath="inventoryData" />
);

export default InventoryConnected;
