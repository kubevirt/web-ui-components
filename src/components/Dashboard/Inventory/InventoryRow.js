import React from 'react';
import PropTypes from 'prop-types';

import { prefixedId } from '../../../utils';

import { InventoryItemStatus } from './InventoryItemStatus';
import { InlineLoading } from '../../Loading';

const InventoryRowTitle = ({ title, count }) => (
  <div className="kubevirt-inventory__row-title">{count != null ? `${count} ${title}` : title}</div>
);

InventoryRowTitle.defaultProps = {
  count: null,
};

InventoryRowTitle.propTypes = {
  title: PropTypes.string.isRequired,
  count: PropTypes.number,
};

export const InventoryRow = ({ title, count, ...rest }) => (
  <div id={prefixedId('inventory', title.toLowerCase())} className="kubevirt-inventory__row">
    <InventoryRowTitle count={count} title={title} />
    {count != null ? <InventoryItemStatus {...rest} key="status" /> : <InlineLoading />}
  </div>
);

InventoryRow.defaultProps = {
  ...InventoryRowTitle.defaultProps,
  ...InventoryItemStatus.defaultProps,
};

InventoryRow.propTypes = {
  ...InventoryRowTitle.propTypes,
  ...InventoryItemStatus.propTypes,
};
