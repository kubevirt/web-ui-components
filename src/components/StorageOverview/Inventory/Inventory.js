import React from 'react';
import PropTypes from 'prop-types';

import {
  DashboardCard,
  DashboardCardBody,
  DashboardCardHeader,
  DashboardCardTitle,
  DashboardCardTitleHelp,
} from '../../Dashboard/DashboardCard';
import { StorageOverviewContext } from '../StorageOverviewContext';
import { mapNodesToProps, mapPvcsToProps, mapPvsToProps } from '../../Dashboard/Inventory/utils';
import diskStatsToProps from './diskInventoryUtils';
import { InventoryRow } from '../../Dashboard/Inventory/InventoryRow';
import { InlineLoading } from '../../Loading';

const InventoryBody = ({ nodes, pvs, pvcs, diskStats, LoadingComponent }) => (
  <React.Fragment>
    <InventoryRow title="Nodes" {...mapNodesToProps(nodes)} LoadingComponent={LoadingComponent} />
    <InventoryRow title="Disks" {...diskStatsToProps(diskStats)} LoadingComponent={LoadingComponent} />
    <InventoryRow title="PVs" {...mapPvsToProps(pvs)} LoadingComponent={LoadingComponent} />
    <InventoryRow title="PVCs" {...mapPvcsToProps(pvcs)} LoadingComponent={LoadingComponent} />
  </React.Fragment>
);

InventoryBody.defaultProps = {
  nodes: undefined,
  diskStats: undefined,
  pvs: undefined,
  pvcs: undefined,
  LoadingComponent: InlineLoading,
};

InventoryBody.propTypes = {
  nodes: PropTypes.array,
  diskStats: PropTypes.object,
  pvs: PropTypes.array,
  pvcs: PropTypes.array,
  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
};

export const Inventory = props => (
  <DashboardCard>
    <DashboardCardHeader>
      <DashboardCardTitle>Inventory</DashboardCardTitle>
      <DashboardCardTitleHelp>help for inventory</DashboardCardTitleHelp>
    </DashboardCardHeader>
    <DashboardCardBody>
      <InventoryBody {...props} />
    </DashboardCardBody>
  </DashboardCard>
);

Inventory.defaultProps = {
  ...InventoryBody.defaultProps,
};

Inventory.propTypes = {
  ...InventoryBody.propTypes,
};

export const InventoryConnected = () => (
  <StorageOverviewContext.Consumer>{props => <Inventory {...props} />}</StorageOverviewContext.Consumer>
);
