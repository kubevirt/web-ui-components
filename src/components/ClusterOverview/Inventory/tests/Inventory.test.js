import React from 'react';
import { render } from 'enzyme';

import { Inventory } from '../Inventory';
import { default as InventoryFixtures } from '../fixtures/Inventory.fixture';

const testInventoryOverview = () => <Inventory {...InventoryFixtures[0].props} />;

describe('<Inventory />', () => {
  it('renders correctly', () => {
    const component = render(testInventoryOverview());
    expect(component).toMatchSnapshot();
  });
});
