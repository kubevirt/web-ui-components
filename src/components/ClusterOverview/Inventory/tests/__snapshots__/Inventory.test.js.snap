// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Inventory /> renders correctly 1`] = `
<div
  class="card-pf kubevirt-dashboard__card"
>
  <div
    class="card-pf-heading kubevirt-dashboard__card-heading"
  >
    <h2
      class="card-pf-title"
    >
      Cluster inventory
    </h2>
    <button
      class="btn btn-link"
      type="button"
    >
      <span
        aria-hidden="true"
        class="fa fa-info-circle fa-2x"
      />
    </button>
  </div>
  <div
    class="card-pf-body"
  >
    <div
      class="kubevirt-inventory__row"
      id="inventory-nodes"
    >
      <div
        class="kubevirt-inventory__row-title"
      >
        1 Nodes
      </div>
      <div
        class="kubevirt-inventory__row-status"
      >
        <div
          class="kubevirt-inventory__row-status-item"
        >
          <span
            aria-hidden="true"
            class="fa fa-check-circle fa-2x kubevirt-inventory__row-status-item-icon--ok"
          />
          <span
            class="kubevirt-inventory__row-status-item-text"
          >
            1
          </span>
        </div>
      </div>
    </div>
    <div
      class="kubevirt-inventory__row"
      id="inventory-pvcs"
    >
      <div
        class="kubevirt-inventory__row-title"
      >
        5 PVCs
      </div>
      <div
        class="kubevirt-inventory__row-status"
      >
        <div
          class="kubevirt-inventory__row-status-item"
        >
          <span
            aria-hidden="true"
            class="fa fa-check-circle fa-2x kubevirt-inventory__row-status-item-icon--ok"
          />
          <span
            class="kubevirt-inventory__row-status-item-text"
          >
            5
          </span>
        </div>
      </div>
    </div>
    <div
      class="kubevirt-inventory__row"
      id="inventory-pods"
    >
      <div
        class="kubevirt-inventory__row-title"
      >
        1 Pods
      </div>
      <div
        class="kubevirt-inventory__row-status"
      >
        <div
          class="kubevirt-inventory__row-status-item"
        >
          <span
            aria-hidden="true"
            class="fa fa-check-circle fa-2x kubevirt-inventory__row-status-item-icon--ok"
          />
          <span
            class="kubevirt-inventory__row-status-item-text"
          >
            1
          </span>
        </div>
      </div>
    </div>
    <div
      class="kubevirt-inventory__row"
      id="inventory-vms"
    >
      <div
        class="kubevirt-inventory__row-title"
      >
        2 VMs
      </div>
      <div
        class="kubevirt-inventory__row-status"
      >
        <div
          class="kubevirt-inventory__row-status-item"
        >
          <span
            aria-hidden="true"
            class="fa fa-check-circle fa-2x kubevirt-inventory__row-status-item-icon--ok"
          />
          <span
            class="kubevirt-inventory__row-status-item-text"
          >
            2
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
